manta-crypto/src/eclair/num.rs

```diff
@@ -180,6 +257,90 @@ where
     }
 }
 
+impl<T, const BITS: usize, COM> Zero<COM> for UnsignedInteger<T, BITS>
+where
+    T: Zero<COM>,
+{
+    type Verification = T::Verification;
+
+    #[inline]
+    fn zero(compiler: &mut COM) -> Self {
+        Self::new_unchecked(T::zero(compiler))
+    }
+
+    #[inline]
+    fn is_zero(&self, compiler: &mut COM) -> Self::Verification {
+        self.0.is_zero(compiler)
+    }
+}
+
+impl<T, const BITS: usize, COM> One<COM> for UnsignedInteger<T, BITS>
+where
+    T: One<COM>,
+{
+    type Verification = T::Verification;
+
+    #[inline]
+    fn one(compiler: &mut COM) -> Self {
+        Self::new_unchecked(T::one(compiler))
```

Comment:
This is slightly more subtle. I'm not sure what to do here since its technically false that `1 < 2^0` so if `BITS` is set to zero then this is implementation is weird. One thing we could add is that `BITS > 0` whenever a compiler implements `AssertWithinBitRange`.

Reply:
I added a comment for this.